Formula/kvrocks.rb

```diff
@@ -0,0 +1,28 @@
+class Kvrocks < Formula
+  desc "Key-value NoSQL database based on RocksDB and compatible with Redis protocol"
+  homepage "https://github.com/bitleak/kvrocks/"
+  url "https://github.com/bitleak/kvrocks.git",
+    tag:      "v2.0.1",
+    revision: "5f41145adb25a79706a68d233a28a0f633ef255f"
+
+  license "BSD-3-Clause"
+
+  head "https://github.com/bitleak/kvrocks.git", branch: "unstable"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build
+  depends_on "libtool" => :build
+
+  def install
+    mkdir_p "_build"
+    system "sh", "build.sh", "_build"
+    bin.install "_build/kvrocks"
+    etc.install "kvrocks.conf"
+  end
+
+  test do
+    assert_match "/Version: 999\.999\.999.*/", shell_output("#{bin}/kvrocks -v 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Has added the new test case